presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetHiveRecordCursor.java

```diff
@@ -771,78 +780,183 @@ public void end()
         void afterValue();
     }
 
+    private interface BlockFieldConverter extends BlockConverter
```

Comment:
This is really a block converter that keeps track of the index and type of the field, so maybe we should find a better name (`block field` sounds weird). Also there is a `Pair` class that already wraps the field and type info, so we can use that here too.